pkgs/applications/misc/gopass-ui/node-composition.nix

```diff
@@ -0,0 +1,17 @@
+# This file has been generated by node2nix 1.7.0. Do not edit!
+
+{pkgs ? import <nixpkgs> {
+    inherit system;
+  }, system ? builtins.currentSystem, nodejs ? pkgs."nodejs-12_x"}:
+
+let
+  nodeEnv = import ../../../development/node-packages/node-env.nix {
+    inherit (pkgs) stdenv python2 utillinux runCommand writeTextFile;
+    inherit nodejs;
+    libtool = if pkgs.stdenv.isDarwin then pkgs.darwin.cctools else null;
```

Comment:
Failed to build on darwin:
```sh
$ nix-build -A gopass-ui --show-trace
error: while evaluating the attribute 'buildPhase' of the derivation 'gopass-ui-0.7.0' at /Users/sikmir/Projects/nixpkgs/pkgs/applications/misc/gopass-ui/default.nix:30:3:
while evaluating the attribute 'buildInputs' of the derivation 'node_gopass-ui-0.7.0' at /Users/sikmir/Projects/nixpkgs/pkgs/development/node-packages/node-env.nix:397:7:
while evaluating 'getOutput' at /Users/sikmir/Projects/nixpkgs/lib/attrsets.nix:464:23, called from undefined position:
while evaluating anonymous function at /Users/sikmir/Projects/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:142:17, called from undefined position:
attribute 'darwin' missing, at /Users/sikmir/Projects/nixpkgs/pkgs/applications/misc/gopass-ui/node-composition.nix:11:44
```

Reply:
@sikmir: Thanks. Should be alright now. Can you re-check (on nixos here).

Reply:
```
error: undefined variable 'darwin' at /Users/sikmir/Projects/nixpkgs/pkgs/applications/misc/gopass-ui/node-dependencies.nix:12:13
```

Reply:
@sikmir: Thanks again. `default.nix` was missing that `darwin` argument. I also rebased on top of lastest `master` as it seem there was a minor change in `nodePackages` naming.

Reply:
Failed, see https://pastebin.com/dB3UPjrQ.

Reply:
Seems that it cannot find `xcodebuild`. I'm not quite familiar with the nix on darwin workflow. Is there any prerequisites on installing xcode?

Reply:
Found the following entries in the manual:

 -  [NixOS - Nixpkgs manual - 10.1. Darwin (macOS)](https://nixos.org/nixpkgs/manual/#sec-darwin)

     > The package xcbuild can be used to build projects that really depend on Xcode. However, this replacement is not 100% compatible with Xcode and can occasionally cause issues. 

 -  [NixOS - Nixpkgs manual - 15.11. iOS](https://nixos.org/nixpkgs/manual/#ios)

Reply:
Might experiment by adding the following to the nativeBuildInputs:

`    ++ stdenv.lib.optionals stdenv.isDarwin [ xcbuild darwin.DarwinTools ];`

Reply:
Here's what I would try first:

```diff


diff --git a/pkgs/applications/misc/gopass-ui/default.nix b/pkgs/applications/misc/gopass-ui/default.nix
index 6ecb0aa1bd6..f82c6aa3945 100644
--- a/pkgs/applications/misc/gopass-ui/default.nix
+++ b/pkgs/applications/misc/gopass-ui/default.nix
@@ -23,6 +23,7 @@
 , python2
 , utillinux
 , darwin
+, xcbuild
 , bash
 } @ args:
 
@@ -51,7 +52,7 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     electron_7
-  ];
+  ] ++ stdenv.lib.optionals stdenv.isDarwin [ xcbuild darwin.DarwinTools ];
 
   buildPhase = ''
     dev_node_modules="${nodeDeps}/lib/node_modules/gopass-ui/node_modules"
diff --git a/pkgs/applications/misc/gopass-ui/node-dependencies.nix b/pkgs/applications/misc/gopass-ui/node-dependencies.nix
index cbe810936c7..7fa458ee8e4 100644
--- a/pkgs/applications/misc/gopass-ui/node-dependencies.nix
+++ b/pkgs/applications/misc/gopass-ui/node-dependencies.nix
@@ -42,7 +42,7 @@ let
   buildInputs = [
     # Patching various `package.json` files.
     jq
-  ];
+  ] ++ stdenv.lib.optionals stdenv.isDarwin [ xcbuild darwin.DarwinTools ];
 
   preRebuild = ''
     post_install_package_scripts_remove() {


```

Reply:
Do you want to help with supporting darwin?

Reply:
I can push this tentative patch and then provide assistance with further experimentation? Would you like to be added as maintainer?

Reply:
Done for pushing the tentative patch.

Reply:
Yes, I can help with supporting/maintaining darwin.

Reply:
Alright, I added you as maintainer. Thanks.